### DISCUSSION TITLE
Osmosis-merged ways not routable?

### DISCUSSION THREAD

**Discussion by alexanderzatko** [Project-OSRM/osrm-backend#6816] [Q&A] [answered]
My workflow requires appending new ways to existing .pbf. I am using Osmosis to combine .osm file with .pbf file to produce a new .pbf file. After appending and starting OSRM, using the **tile** service, I see the new ways were added to the routing graph, but fail to find a route over the newly-added ways when querying OSRM directly, or through the osrm frontend.

here is a piece of my .osm file:

```
...
<node timestamp="2023-10-23T09:26:18+02:00" version="1" id="901698046036" action="modify" lat="48.773767246464" lon="19.020828542855" />
<node timestamp="2023-10-23T09:26:18+02:00" version="1" id="901698046037" action="modify" lat="48.773833502256" lon="19.021076080397" />
<way timestamp="2023-10-23T09:26:18+02:00" version="1" id="901698055978" action="modify">
<nd ref="901698045978" />
<nd ref="901698045979" />
<nd ref="901698045980" />
<nd ref="901698045981" />
<nd ref="901698045982" />
<nd ref="901698045983" />
<tag k="highway" v="track" />
</way>
...
```

my .lua profile allows routing on highway=track.

Any ideas what could I try to troubleshoot the issue?

**Comment by nilsnolde** [comment, marked as answer]
You’ll need to make sure that the new ways are also _connected_ to the rest of the network. Meaning, connecting ways need to have the same node ID referenced.

**Comment by alexanderzatko** [comment]
It did cross my mind, but I also thought the appended ways should be routable independently of the rest, which is not the case. Thank you for the suggestion anyways, I will try that.

**Comment by alexanderzatko** [comment]
This advice worked for me. My workflow now consists of opening JOSM, downloading the area of interest, adding the new ways and submitting the changes to the osm db. Then I get the IDs of the new ways and the ways the new ways connect to in newly-added nodes (if any were added), delete the data layer and get one-by-one all of the newly added ways into a new blank data layer. Then I save the work into .osm file. Such file is then combined with mu local .pbf. Works every time.
